### PR TITLE
Documentation linkcheck run, fixing some links.

### LINKED ISSUE #4359: [DOCS] Use HTTPS
Reference:
 - https://httpsiseasy.com
 - https://letsencrypt.org

### PR DESCRIPTION
**Pull request by nyov**
Closes #4359

### CODE REVIEW

File: docs/intro/install.rst [hunk 1]
```diff
@@ -65,8 +65,8 @@ please refer to their respective installation instructions:
 * `lxml installation`_
 * `cryptography installation`_
 
-.. _lxml installation: http://lxml.de/installation.html
-.. _cryptography installation: https://cryptography.io/en/latest/installation/
+.. _lxml installation: https://lxml.de/installation.html
+.. _cryptography installation: https://cryptography.io/en/latest/en/latest/installation/
```
Comment (nyov) [inline docs/intro/install.rst]: The script has some weird string replace issues. Looks like I missed this one.

File: docs/topics/selectors.rst [hunk 2]
```diff
@@ -35,12 +35,12 @@ defines selectors to associate those styles with specific HTML elements.
     in speed and parsing accuracy to lxml.
 
 .. _BeautifulSoup: https://www.crummy.com/software/BeautifulSoup/
-.. _lxml: http://lxml.de/
+.. _lxml: https://lxml.de/
 .. _ElementTree: https://docs.python.org/2/library/xml.etree.elementtree.html
-.. _cssselect: https://pypi.python.org/pypi/cssselect/
-.. _XPath: https://www.w3.org/TR/xpath
+.. _cssselect: https://pypi.org/project/cssselect/
```
Comment (nyov) [inline docs/topics/selectors.rst]: `_cssselect` is defined multiple times. Here, earlier as `_cssselect: https://github.com/scrapy/cssselect/`, and apparently in parsel, which gives a weird (for me) build conflict: https://github.com/scrapy/parsel/issues/185

What might be the best way to resolve this?

Comment (Gallaecio) [reply docs/topics/selectors.rst]: One way may be to use InterSphinx (https://github.com/scrapy/scrapy/blob/master/docs/conf.py#L282) for CSS select, and have those two references use something like :doc:`cssselect <cssselect:index>` instead.

Comment (nyov) [reply docs/topics/selectors.rst]: That seems sensible. I don't think anyone would mind the link change from the pypi & github pages to the docs, there.

File: docs/conf.py [hunk 3]
```diff
@@ -287,6 +287,7 @@
     'tox': ('https://tox.readthedocs.io/en/latest', None),
     'twisted': ('https://twistedmatrix.com/documents/current', None),
     'twistedapi': ('https://twistedmatrix.com/documents/current/api', None),
+    'cssselect': ('https://cssselect.readthedocs.io/en/latest', None),
```
Comment (Gallaecio) [inline docs/conf.py]: :lipstick: A-Z

Comment (nyov) [reply docs/conf.py]: ............................................ah! Well that took me embarrassingly long to parse.

Comment (nyov) [reply docs/conf.py]: ............................................ah! Well that took me embarrassingly long to parse.

**Comment by kmike** [comment]
nice, thanks for the cleanup!

**Pull request merged**